File: 1768. Merge Strings Alternately/1768. Merge Strings Alternately.java
class Solution {

  public String mergeAlternately(String word1, String word2) {
    int l1 = word1.length();
    int l2 = word2.length();
    int maxLength = Math.max(l1, l2);
    StringBuilder merged = new StringBuilder();

    for (int i = 0; i < maxLength; i++) {
      if (i < l1) {
        merged.append(word1.charAt(i));
      }

      if (i < l2) {
        merged.append(word2.charAt(i));
      }
    }

    return merged.toString();
  }
}

File: 1768. Merge Strings Alternately/1768. Merge Strings Alternately.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {string}
 */
var mergeAlternately = function (word1, word2) {
  const l1 = word1.length;
  const l2 = word2.length;
  const maxLength = Math.max(l1, l2);
  const merged = [];

  for (let i = 0; i < maxLength; i++) {
    if (i < l1) {
      merged.push(word1[i]);
    }

    if (i < l2) {
      merged.push(word2[i]);
    }
  }

  return merged.join("");
};
